src/test/java/org/houseflys/jdbc/PreparedStatementITest.java

```diff
@@ -0,0 +1,158 @@
+        withNewConnection(new WithConnection() {
+            @Override
+            public void apply(Connection connection) throws Exception {
+                PreparedStatement preparedStatement = connection.prepareStatement("SELECT ?,?");
+
+                preparedStatement.setString(1, "test1");
+                preparedStatement.setString(2, "test2");
+                ResultSet rs = preparedStatement.executeQuery();
+                Assert.assertTrue(rs.next());
+                Assert.assertEquals(rs.getString(1), "test1");
+                Assert.assertEquals(rs.getString(2), "test2");
+                Assert.assertFalse(rs.next());
+            }
+        });
+    }
+
+    @Test
+    public void successfullyNullable() throws Exception {
+        withNewConnection(new WithConnection() {
+            @Override
+            public void apply(Connection connection) throws Exception {
+                PreparedStatement preparedStatement = connection.prepareStatement("SELECT arrayJoin([?,?])");
+
+                preparedStatement.setString(1, null);
+                preparedStatement.setString(2, "test2");
+                ResultSet rs = preparedStatement.executeQuery();
+                Assert.assertTrue(rs.next());
+                Assert.assertEquals(rs.getString(1), "");
+                Assert.assertTrue(rs.wasNull());
+                Assert.assertTrue(rs.next());
+                Assert.assertEquals(rs.getString(1), "test2");
+                Assert.assertFalse(rs.next());
+            }
+        });
+    }
+
+    @Test
+    public void successfullyDate() throws Exception {
+        withNewConnection(new WithConnection() {
+            @Override
+            public void apply(Connection connection) throws Exception {
+                PreparedStatement preparedStatement = connection.prepareStatement("SELECT toDate(?)");
+
+                long now = System.currentTimeMillis();
+                preparedStatement.setDate(1, new Date(now));
+                ResultSet rs = preparedStatement.executeQuery();
+                Assert.assertTrue(rs.next());
+                Assert.assertEquals(rs.getDate(1).getTime(),
+                    now / TimeUnit.DAYS.toMillis(1) * TimeUnit.DAYS.toMillis(1));
+                Assert.assertFalse(rs.next());
+            }
+        });
+    }
+
+    @Test
+    public void successfullyInsertData() throws Exception {
+        withNewConnection(new WithConnection() {
+            @Override
+            public void apply(Connection connection) throws Exception {
+                Statement statement = connection.createStatement();
+
+                statement.execute("DROP TABLE IF EXISTS test");
+                statement.execute("CREATE TABLE test(id UInt8)ENGINE = Log");
+
+                PreparedStatement preparedStatement =
+                    connection.prepareStatement("INSERT INTO test VALUES(1)(?)(3)(?)");
+
+                preparedStatement.setByte(1, (byte) 2);
```

Comment:
May be it could be compatible with   ` preparedStatement.setInt(1, 2);`

In `DataTypeInt8`,  we could use some transformers like 

```
  @Override
    public void serializeBinary(Object data, BinarySerializer serializer) throws SQLException, IOException {
        Validate.isTrue(data instanceof Number,
            "Can't serializer " + data.getClass().getSimpleName() + " With ByteDataTypeSerializer.");
        serializer.writeByte(((Number) data).byteValue());
    }
```

Reply:
We should report this overflow problem to the user.